pre_release.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+usage() { echo "Usage: $0 -t <tag>" 1>&2; exit 1; }
+
+while getopts ":t:" o; do
+    case "${o}" in
+        t)
+            tag=${OPTARG}
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+if [ ! -z "${t}" ]; then
+    usage
+fi
+
+#validate tag
+semver_regex="^(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(\\-[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?(\\+[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?$"
+if [[ ${tag} =~ ${semver_regex} ]]; then
+    echo "${tag} is valid semver tag"
+else
+    echo "Error: ${tag} is not a valid semver tag. Exiting"
+    exit -1
+fi
+
+#ensure tag doesn't exits
+if [[ $(git tag --list ${tag}) ]]; then
+   echo "Error: Tag ${tag} already exists. Exiting"
+   exit -1
+fi
+
+if ! git diff --quiet; then \
```

Comment:
Does this mean `git diff --quiet` exists with `0` if there is any local change?

Reply:
This is checking the (standard ) output from the command, and not the return code (`$?`) of the command  So if there is any local changes, the command would print those changes. The return code would remain 0 irrespective of any local changes or not.